Source/Engine/Utilities/Extensions.cs

```diff
@@ -404,5 +238,184 @@ public static void Shuffle<T>(this Random random, IList<T> collection)
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <param name="weight">Normalized value that determines the chance to return true.</param>
+        /// <returns>A <see cref="bool"/> thats either true or false.</returns>
+        public static bool NextBool(this Random random, float weight = 0.5f) => weight >= NextFloat(random);
+
+        /// <summary>
+        /// Generates a random <see cref="byte"/> value between min and max.
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <param name="min">The minimum value.</param>
+        /// <param name="max">The maximum value.</param>
+        /// <returns>A <see cref="byte"/> between min and max.</returns>
+        public static byte NextByte(this Random random, byte min = 0, byte max = 2)
+        {
+            return (byte)random.Next(min, max);
+        }
+
+        /// <summary>
+        /// Generates a random <see cref="float"/>  value between min and max.
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <param name="min">The minimum value.</param>
+        /// <param name="max">The maximum value.</param>
+        /// <returns>A random <see cref="float"/> between min and max.</returns>
+        public static float NextFloat(this Random random, float min = 0.0f, float max = 2.0f)
+        {
+            return (float)random.NextDouble() * (max - min) + min;
+        }
+
+        /// <summary>
+        /// Generates a random <see cref="Quaternion"/>.
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <param name="randomRoll">Should the roll value be randomized.</param>
+        /// <returns>A random <see cref="Quaternion"/>.</returns>
+        public static Quaternion NextQuaternion(this Random random, bool randomRoll = false)
+        {
+
+            return Quaternion.Euler(
+                NextFloat(random, -180, 180),
+                NextFloat(random, -180, 180),
+                randomRoll ? NextFloat(random, -180, 180) : 0);
+        }
+
+        /// <summary>
+        /// Generates a uniformly distributed random unit length vector point on a unit sphere.
+        /// </summary>
+        /// <param name="random">An instance of <see cref="Random"/>.</param>
+        /// <returns>A random <see cref="Vector3"/>.</returns>
+        public static Vector3 NextVector3(this Random random)
+        {
+            Vector3 output;
+
+            do
+            {
+                // Create random float with a mean of 0 and deviation of ±1
+                output.X = NextFloat(random) * 2.0f - 1.0f;
+                output.Y = NextFloat(random) * 2.0f - 1.0f;
+                output.Z = NextFloat(random) * 2.0f - 1.0f;
+
+            } while (output.LengthSquared > 1 || output.LengthSquared < 1e-6f);
```

Comment:
You should create a new local variable to store the `output.LengthSquared` result, to prevent calculating it more than one time.
The number `1e-6f` could be replaced with the constant variable defined in `Mathf.Epsilon`.

Reply:
Noted & done.